### PR TITLE
:seedling: Update actions for 1.5 and make names consistent

### PR DESCRIPTION
**Pull request by killianmuldoon**
Update actions for the release-1.5 branch - this was missed as part of creating the release branch initially, but is documented in https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#tasks-2

This PR also updates the names of the github actions run on the repo to clarify which are run on PRs and which are periodic weekly jobs.

### CODE REVIEW

File: .github/workflows/pr-golangci-lint.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: golangci-lint
```
Comment (sbueringer) [inline .github/workflows/pr-golangci-lint.yml:1]: While we're at it. Let's please use `.yaml` consistently (there are probably references to the action file names somewhere)

(please also update jobs.md)

**Review by sbueringer — COMMENTED** [review]
Thx for catching this

**Review by sbueringer — COMMENTED** [review]
Thx for catching this

**Comment by sbueringer** [comment]
@killianmuldoon we're going to need something similar for CAPV :)

**Comment by killianmuldoon** [comment]
>we're going to need something similar for CAPV :)

I'll take a look

**Pull request merged**